pkg/trafficpolicy/egress_types.go

```diff
@@ -72,6 +70,5 @@ type EgressHTTPRoutingRule struct {
 	Route RouteWeightedClusters
 
 	// AllowedDestinationIPRanges defines the destination IP ranges allowed for the `Route` defined in the routing rule.
-	// Stores string type
-	AllowedDestinationIPRanges mapset.Set
+	AllowedDestinationIPRanges []string
```

Comment:
List of strings is a better type than a set of `interface{}`
I wonder if it would be too cumbersome to do `net.IP{}`

Reply:
Yes, I initially left this as a set of `interface{}` and it got ugly to the extent where bugs were creeping in where type assertions were not being enforced each time the set was used, and involved repeated conversions between`[]string{} <-> set(interface{})`. Changing it to `[]string` did prevent such issues, with the usage of `mapset` limited to within the function.

I contemplated using `net.IPNet` instead, from which the IP network and prefix length can be retrieved, essentially a struct representation of `a.b.c.d/x`. I'll consider this again when the IP ranges associated with routes get mapped to XDS configurations.